### PR TITLE
[GLUTEN-5512][CH] Fix the incorrect transformer for the round function with the decimal data type

### LINKED ISSUE #5512: [CH] Fix the incorrect transformer for the round function with the decimal data type
### Backend

CH (ClickHouse)

### Bug description

[CH] Fix the incorrect transformer for the round function with the decimal data type.

When transforming the round function with the decimal data type, it only transformers the `child` expression, but missing the `scale` expression, which will lead to the incorrect results when executing the `round(decimal_data, x).

### Spark version

None

### Spark configurations

_No response_

### System information

_No response_

### Relevant logs

_No response_

### PR DESCRIPTION
**Pull request by zzcclp**
## What changes were proposed in this pull request?

[CH] Fix the incorrect transformer for the round function with the decimal data type.

When transforming the round function with the decimal data type, it only transformers the `child` expression, but missing the `scale` expression, which will lead to the incorrect results when executing the `round(decimal_data, x)`.

Close #5512.

(Fixes: #5512)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by zzcclp** [comment]
@loneylee please help to review.

File: gluten-core/src/main/scala/org/apache/gluten/backendsapi/SparkPlanExecApi.scala [hunk 1]
```diff
@@ -419,7 +419,7 @@ trait SparkPlanExecApi {
       substraitExprName: String,
       child: ExpressionTransformer,
       original: Round): ExpressionTransformer = {
-    GenericExpressionTransformer(substraitExprName, Seq(child), original)
+    DecimalRoundTransformer(substraitExprName, child, original)
```
Comment (rui-mo) [inline gluten-core/src/main/scala/org/apache/gluten/backendsapi/SparkPlanExecApi.scala:422]: It seems we can remove the API `genDecimalRoundTransformer` directly as it has the same implementation as Velox backend.

**Review by rui-mo — COMMENTED** [review]
Thanks.

File: gluten-core/src/main/scala/org/apache/gluten/backendsapi/SparkPlanExecApi.scala (hunk 1 above)
Comment (zzcclp) [reply gluten-core/src/main/scala/org/apache/gluten/backendsapi/SparkPlanExecApi.scala]: Done

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Comment by loneylee** [comment]
LGTM

**Review by rui-mo — APPROVED** [review]
Thanks.

**Review by baibaichen — APPROVED** [review]
LGTM

**Comment by zzcclp** [comment]
Run Gluten Clickhouse CI

**Pull request merged**